etna/models/nn/tft_native/layers.py

```diff
@@ -0,0 +1,293 @@
+from typing import Dict
+from typing import Optional
+from typing import Tuple
+
+from etna import SETTINGS
+
+if SETTINGS.torch_required:
+    import torch
+    import torch.nn as nn
+
+
+class GatedLinearUnit(nn.Module):
+    """Gated Linear Unit."""
+
+    def __init__(self, input_size: int, output_size: int, dropout: float = 0.1) -> None:
+        """Init Gated Linear Unit.
+
+        Parameters
+        ----------
+        input_size:
+            input size of the feature representation
+        output_size:
+            output size of the feature representation
+        dropout:
+            dropout rate
+        """
+        super().__init__()
+
+        self.input_size = input_size
+        self.output_size = output_size
+        self.dropout = nn.Dropout(dropout)
+        self.activation_fc = nn.Linear(self.input_size, self.output_size)
+        self.sigmoid = nn.Sigmoid()
+        self.gated_fc = nn.Linear(self.input_size, self.output_size)
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        """Forward pass.
+
+        Parameters
+        ----------
+        x:
+            batch of data with shapes (batch_size, num_timestamps, input_size)
+
+        Returns
+        -------
+        :
+            output batch of data with shapes (batch_size, num_timestamps, output_size)
+        """
+        x = self.dropout(x)
+        a = self.activation_fc(x)
+        b = self.sigmoid(self.gated_fc(x))
+        x = torch.mul(a, b)
+        return x
+
+
+class GateAddNorm(nn.Module):
+    """Gated Add&Norm layer."""
+
+    def __init__(self, input_size: int, output_size: int, dropout: float = 0.1) -> None:
+        """Init Add&Norm layer.
+
+        Parameters
+        ----------
+        input_size:
+            input size of the feature representation
+        output_size:
+            output size of the feature representation
+        dropout:
+            dropout rate
+        """
+        super().__init__()
+
+        self.input_size = input_size
+        self.output_size = output_size
+        self.dropout = dropout
+
+        self.glu = GatedLinearUnit(input_size=self.input_size, output_size=self.output_size, dropout=self.dropout)
+        self.norm = nn.LayerNorm(self.output_size)
+
+    def forward(self, x: torch.Tensor, residual: torch.Tensor) -> torch.Tensor:
+        """Forward pass.
+
+        Parameters
+        ----------
+        x:
+            batch of data with shapes (batch_size, num_timestamps, input_size)
+        residual:
+            batch of data passed through skip connection with shapes (batch_size, num_timestamps, output_size)
+        Returns
+        -------
+        :
+            output batch of data with shapes (batch_size, num_timestamps, output_size)
+        """
+        x = self.glu(x)
+        x = self.norm(x + residual)
+        return x
+
+
+class GatedResidualNetwork(nn.Module):
```

Comment:
Actually the reference implementation here should be implementation from google [see](https://github.com/google-research/google-research/blob/master/tft/libs/tft_model.py)

Reply:
What it means?

Reply:
It means that if you use some implementation as reference it is better to be this one